spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/BasicRelationalPersistentProperty.java

```diff
@@ -173,14 +188,22 @@ public int getSqlType() {
 
 	@Override
 	public String getReverseColumnName() {
-		return columnName.get().orElseGet(() -> context.getNamingStrategy().getReverseColumnName(this));
+		return collectionIdColumnName.get().orElseGet(
```

Comment:
This resolution (relation to `columnName`) should rather go into `collectionIdColumnName` resolution to keep the fallback code more closely together.